### DISCUSSION TITLE
Is possible to declare any attribute to inform that a (bool)property handles the null check for other given property?

### DISCUSSION THREAD

**Discussion by sn4k3** [dotnet/runtime#89352] [General] [answered]
Take this code as example:

```C#
#nullable enable

public string? FilePath { get; private set; }

public bool IsFileLoaded => !string.IsNullOrWhiteSpace(FilePath);

public void ReloadFile()
{
   if (!IsFileLoaded) return;
   var file = File.OpenRead(FilePath);
}
```

The compiler will warn about `FilePath` not being checked for null before access it, but it is on the previous if statement. Visual Studio intellisense is also not able to determine such to skip the code advice.
Is there any solution for this? If not, is it possible to implement some attribute that can handle such? eg: 

```C#
[IsNotNullCheck(nameof(FilePath))]
public bool IsFileLoaded => !string.IsNullOrWhitespace(FilePath);

[IsNullCheck(nameof(FilePath))]
public bool IsFileNotLoaded => string.IsNullOrWhitespace(FilePath);
```

So, on this case if use `IsFileLoaded` the nullable warning should be ignored since it's handled.
Of course, this is also unnecessary as compiler can query the properties and to see if the logic is handling the null of any property.

**Comment by huoyaoyuan** [comment, marked as answer]
You can use `[MemberNotNullWhen]` on the `IsFileLoaded` property.

**Reply by sn4k3** [reply]
Thanks! That solved what I was seeking :)

```C#
[MemberNotNullWhen(true, nameof(FilePath))]
public bool IsFileLoaded => !string.IsNullOrWhiteSpace(FilePath);
```

Would be nice to have Visual Studio suggest such use of attribute when detecting that possible usage.

**Comment by Clockwork-Muse** [comment]
Side note: I wouldn't consider a file "loaded" if only the file name was set.  "Named", maybe.

**Reply by sn4k3** [reply]
Thanks for pointing out, I agree with you that is not the safest even if class is the only thing handling the `FilePath` and `FileStream` (in private). It really depends on how class is designed, instead a check within the `FileStream` would be more bullet-proof and safer.

However, this was just a quick sample that come to my head, I'm not really using that pattern on my code.
